Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -375,6 +375,17 @@ extension StripeCardReaderService: CardReaderService {
 
 // MARK: - Payment collection
 private extension StripeCardReaderService {
+    /// Returns the id of the connected reader, if any
+    ///
+    func readerIDForIntent() -> String? {
+        let connectedReaders = connectedReadersSubject.value
+        guard connectedReaders.count == 1 else {
```

Comment:
Q: So is there a case of many readers where we shouldn't pass any IDs? Does it mean that we're not sure which one of them just used, so we prefer "no data" instead of "possibly wrong data"?

Reply:
We probably should convert `connectedReader` from an array to an optional at some point. From what I understand, only one reader can be connected at a time

Reply:
We should. I can't remember why we made it an array.

Reply:
> So is there a case of many readers where we shouldn't pass any IDs? Does it mean that we're not sure which one of them just used, so we prefer "no data" instead of "possibly wrong data"?

The latter. I didn't want to abort the payment intent in the unlikely event that we think we have no connectedReader. This is just defensive coding that is unlikely to be triggered by real world use.

Only one reader can be connected at a time.

Reply:
> So is there a case of many readers where we shouldn't pass any IDs? Does it mean that we're not sure which one of them just used, so we prefer "no data" instead of "possibly wrong data"?

The latter. I didn't want to abort the payment intent in the unlikely event that we think we have no connectedReader. This is just defensive coding that is unlikely to be triggered by real world use.

Only one reader can be connected at a time.